server/zanata-war/src/main/java/org/zanata/adapter/JsonAdapter.java

```diff
@@ -46,4 +48,9 @@ protected void updateParamsWithDefaults(IParameters params) {
         Parameters p = (Parameters) params;
         p.fromString("extractIsolatedStrings.b=true");
     }
+
+    @Override
+    protected String getIdFor(TextUnit tu, String content, String subDocName) {
+        return HashUtil.generateHash(tu.getName().concat(content));
```

Comment:
I thought maybe the global key would be overkill - but to be honest I'm not confident of this answer. I guess the question is - is it important enough to change it?
Also, could CopyTrans work for re-translating the document after update?